compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParserErrorHandler.java

```diff
@@ -1920,6 +1933,7 @@ protected ParserRuleContext getNextRule(ParserRuleContext currentCtx, int nextLo
             case ARG_LIST_START:
                 return ParserRuleContext.ARG_LIST;
             case ARG_LIST_END:
+            case EXPLICIT_NEW_RHS:
```

Comment:
I don't think we need it here. 
This should be added to all the places where other `TYPE_DESC_IN_**` contexts are added.

Reply:
fixed in 401f175